exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/StatisticsMergeBatch.java

```diff
@@ -0,0 +1,532 @@
+    MaterializedField mf;
+    assert mapVector.getPrimitiveVectors().size() > 0;
+    // Proceed to create output map vector with same name e.g. statcount etc.
+    MajorType mt = mapVector.getField().getType();
+    if (functions.get(field) != null) {
+      mf = MaterializedField.create(functions.get(field), mt);
+    } else {
+      mf = MaterializedField.create(field, mt);
+    }
+    assert !dataSrcVecMap.containsKey(mf);
+    ValueVector vector = TypeHelper.getNewVector(mf, oContext.getAllocator());
+    container.add(vector);
+    MapVector outputMapVector = (MapVector) vector;
+
+    for (ValueVector vv : mapVector) {
+      String fieldName = vv.getField().getLastName();
+      if (!keyList.contains(fieldName)) {
+        throw new UnsupportedOperationException("Unpivot data vector " +
+                field + " contains key " + fieldName + " not contained in key source!");
+      }
+      if (vv.getField().getType().getMinorType() == TypeProtos.MinorType.MAP) {
+        throw new UnsupportedOperationException("Unpivot of nested map is not supported!");
+      }
+      if (field.equals("column")) {
+        outputMapVector.addOrGet(fieldName, vv.getField().getType(), vv.getClass());
+      } else {
+        List<LogicalExpression> args = Lists.newArrayList();
+        LogicalExpression call;
+        //TODO: Something else to access value of col such as emp_id?
+        args.add(SchemaPath.getSimplePath(vv.getField().getPath()));
+        //TODO: Put in the mapVector
+        if (functions.get(field) != null) {
+          call = FunctionCallFactory.createExpression(functions.get(field), args);
+        } else {
+          call = FunctionCallFactory.createExpression(field, args);
+        }
+        //TODO: Is this sufficient to add to new Map?
+        ValueVector vector1 = addMapVector(fieldName, outputMapVector, call);
+        if (collector.hasErrors()) {
+          throw new SchemaChangeException("Failure while materializing expression. "
+                  + collector.toErrorString());
+        }
+      }
+    }
+  }
+
+  @Override
+  protected boolean setupNewSchema() throws SchemaChangeException {
+    container.clear();
+    buildKeyList();
+    buildOutputContainer();
+    return true;
+  }
+
+  @Override
+  protected IterOutcome doWork() {
+    int outRecordCount = incoming.getRecordCount();
+    HashMap<String, Long> nonNullRows = new HashMap<>();
+
+    for (VectorWrapper<?> vw : incoming) {
+      ValueVector vv = vw.getValueVector();
+      if (vv.getField().getType().getMinorType() != TypeProtos.MinorType.MAP) {
+        //We only expect Maps in the incoming. STOP, if this is not the case.
+        return IterOutcome.STOP;
+      }
+      MapVector mapVec = (MapVector) vv;
+      Map<String, ValueHolder> statMap = aggregationMap.get(vv.getField().getLastName());
+
+      for (ValueVector mapElt : mapVec) {
+        if (vv.getField().getLastName().equalsIgnoreCase("column")) {
```

Comment:
Yes, we do not need `.equalsIgnoreCase`. Created `public static final` strings for the hard-coded strings.